test/net/protocol/test_protocol.rb

```diff
@@ -27,6 +27,26 @@ def test_each_crlf_line
     end
   end
 
+  def test_each_message_chunk_boundary
```

Comment:
This test will fail if one uses the old logic for `readuntil` as two chunks will be read together.